##Introduction to phylogenetic tree inference

All phylogenetic analyses include four basic steps: 

	1) data acquisition, 
	2) construction of character matrix (e.g. sequence alignment), 
	3) phylogeny estimation and 
	4) interpretation of phylogenetic inference. 

The objective of this lab exercise is to gain some experience with each of these steps.

##Data acquisition for sequence-based phylogenetic analyses
Researchers typically mix a combination of sequences generated in their own laboratories with sequences generated by others that can be compiled from GenBank and other sequence databases. Visit the following databases and perform term searches and BLAST-based queries:
	
*	http://www.ncbi.nlm.nih.gov
*	http://www.uniprot.org
*	http://www.ensembl.org/info/genome/compara/index.html
*	http://www.phytozome.net
*	http://pfam.xfam.org
*	http://www.treefam.org
*	http://genome.ucsc.edu
*	http://flybase.org

Take notes on how these databases may (or may not) be useful in your research.  Do you know of other useful sequence databases?

##Sequence Alignment
The quality of a sequence alignment can have a major impact on the accuracy of trees estimated through phylogenetic analyses, especially when analyzing divergent sequences (e.g., uncorrected pair-wise distances > 0.6). CLUSTAL (Larkin et al. 2007) is the most widely used progressive alignment program, but over the few years a number of new alignment programs have been published which seem to produce more accurate alignments. Much of the improvement is due to iterative refinement of the guide tree used for the progressive alignment and further refinement of the global alignment through reassessment of local alignment of conserved motifs. In general most alignment programs attempt to group similar sequences and then identify alignments that minimize the number of implied mutations (substitutions, insertions or deletions). Take the following steps to align you sequences using CLUSTALW and MUSCLE (Edgar 2004) and then compare the alignments.

First, download Seaview from: http://doua.prabi.fr/software/seaview (be sure to select x86_64 Linux for the VM)

Open the archive file and change to the Seaview directory:

    tar xzf ~/Downloads/seaview4-64.tgz && cd seaview
 
1. Take a look at the beginning of the FASTA file from yesterday with: `head seqs.faa`.
2. Go to the familiar Uniprot BLAST interface and search the first sequence: 

   http://www.uniprot.org/blast

3. Now, select the first 25 sequences and download as "FASTA" format.

##Phylogenetic Analyses and Interpretation

**Phylogeny estimation**

Today we will estimate Neighbor Joining (distance-based analysis) and Maximum Likelihood trees using a MUSCLE alignment, and Gblocks subset of the MUSCLE alignment. These are two of four widely used paradigms for phylogenetic tree reconstruction. The other two methods are Maximum Parsimony and Bayesian Inference.

The following excercies will all be done in Seaview, and you can launch the application with the command `./seaview` from the Seaview directory.

1. In Seaview, open you FASTA file with "File -> Open"
2. Under "Align" go to "alignment options" and select "MUSCLE"
3. From "Align" select "align all" and wait for MUSCLE to compute the alignment. Notice the messages in the console as the alignment is progressing. Can you figure out what each step in the alignment process is doing?
4. Save the alignment as "seqs_muscle_aln.faa" and inspect the results. How does the alignment look to you?
5. From the Seaview directory, type: 
   `sed -i 's/ .*//g' seqs_muscle_aln.faa` followed by 
    `./Gblocks seqs_muscle_aln.faa`.
6. From the Gblocks menu, hit `g` and Enter, followed by `q` and Enter.
7. In Seaview, analyze the results. Select "File -> Open" again and choose "seqs_muscle_aln.faa-gb". How does this compare with the MUSCLE alignment?
8. Now infer the phylogeny from the Gblocks alignment in Seaview. Select "Tree -> Distance methods" and click "bootstrap" and set this to 500. Now, click Go!
9. **Can you find your sequence in the tree? What is the phylogenetic origin for this group of sequences? Do you have any idea what might explain this finding?**

##Optional
1. Repeat step 2 with Clustal and see if you notice any differences between the alignments. Is the program slower, and does the alignment look as reliable?
2. Repeat step 8 using the Maximum Likelihood methodology. Select "Tree -> PhyML", choose "best of NNI and SPR" and then click Go! As you watch the progress in the console can you figure out what each step is doing?
3. Compare your trees with those estimated by your colleagues. How do trees estimated on the MUSCLE alignments compare with the Gblocks subsets versus Clustal? If there are differences, which trees do you think are more likely to show the true relationships. Can you recall how to test (analytically) hypotheses about different tree topologies?

**References:**
* Castresana, J. 2000. Selection of conserved blocks from multiple alignments for their use in phylogenetic analysis. Molecular Biology  and Evolution 17:540-552
* Edgar R.C. 2004.  MUSCLE: multiple sequence alignment with high accuracy and high throughput.  Nucleic Acids Research 32(5): 1792-1797.
* Larkin M.A., G. Blackshields, N.P. Brown, R. Chenna, P.A. McGettigan, H. McWilliam, F. Valentin, I.M. Wallace, A. Wilm, R. Lopez, J.D. Thompson., T.J. Gibson and D.G. Higgins. 2007.  ClustalW and ClustalX version 2. Bioinformatics 23(21): 2947-2948.